spec/bowling_spec.rb

```diff
@@ -0,0 +1,401 @@
+      allow($stdin).to receive(:gets).and_return(4)
+      bowling.score_input
+      allow($stdin).to receive(:gets).and_return(10)
+      bowling.score_input
+      allow($stdin).to receive(:gets).and_return(4)
+      bowling.score_input
+      allow($stdin).to receive(:gets).and_return(4)
+      bowling.score_input
+      allow($stdin).to receive(:gets).and_return(10)
+      bowling.score_input
+      allow($stdin).to receive(:gets).and_return(4)
+      bowling.score_input
+      allow($stdin).to receive(:gets).and_return(4)
+      bowling.score_input
+      allow($stdin).to receive(:gets).and_return(10)
+      bowling.score_input
+      allow($stdin).to receive(:gets).and_return(4)
+      bowling.score_input
+      allow($stdin).to receive(:gets).and_return(4)
+      bowling.score_input
+      allow($stdin).to receive(:gets).and_return(10)
+      bowling.score_input
+      expect(bowling.input_score_message).to eq 'Please enter your score for round 9, ball 1'
+    end
+
+    it 'goes to the 9th round correctly with 4 spares and 2 strikes' do
+      allow($stdin).to receive(:gets).and_return(4)
+      bowling.score_input
+      allow($stdin).to receive(:gets).and_return(4)
+      bowling.score_input
+      allow($stdin).to receive(:gets).and_return(10)
+      bowling.score_input
+      allow($stdin).to receive(:gets).and_return(4)
+      bowling.score_input
+      allow($stdin).to receive(:gets).and_return(6)
+      bowling.score_input
+      allow($stdin).to receive(:gets).and_return(4)
+      bowling.score_input
+      allow($stdin).to receive(:gets).and_return(6)
+      bowling.score_input
+      allow($stdin).to receive(:gets).and_return(4)
+      bowling.score_input
+      allow($stdin).to receive(:gets).and_return(4)
+      bowling.score_input
+      allow($stdin).to receive(:gets).and_return(4)
+      bowling.score_input
+      allow($stdin).to receive(:gets).and_return(6)
+      bowling.score_input
+      allow($stdin).to receive(:gets).and_return(10)
+      bowling.score_input
+      allow($stdin).to receive(:gets).and_return(4)
+      bowling.score_input
+      allow($stdin).to receive(:gets).and_return(6)
+      bowling.score_input
+      expect(bowling.input_score_message).to eq 'Please enter your score for round 9, ball 1'
+    end
+  end
+
+  describe '#strike' do
+    it 'adds the correct strike bonus in the second round' do
+      allow($stdin).to receive(:gets).and_return(10)
+      bowling.score_input
+      allow($stdin).to receive(:gets).and_return(4)
+      bowling.score_input
+      allow($stdin).to receive(:gets).and_return(4)
+      bowling.score_input
+      expect(bowling.total_score).to eq 26
+    end
+
+    it 'continues to add the bonus with multiple strikes' do
+      12.times do
+        allow($stdin).to receive(:gets).and_return(10)
+        bowling.score_input
+        p bowling.bonus_score
+      end
+      expect(bowling.total_score).to eq 300
+    end
+
+    it 'calculates a lot of stuff' do
```

Comment:
:wink: Yes, and was the it message here helpful during your TDD debugging process?